bitvm/src/signatures/wots_api.rs

```diff
@@ -0,0 +1,150 @@
+                    let mut sigs_vec: Vec<([u8; 20], u8)> = Vec::new();
+                    for i in (0..sigs.len()).step_by(2) {
+                        let preimage: [u8; 20] = if sigs[i].len() == 0 {
+                            [0; 20]
+                        } else {
+                            sigs[i].try_into().unwrap()
+                        };
+                        let digit_arr: [u8; 1] = if sigs[i + 1].len() == 0 {
+                            [0]
+                        } else {
+                            sigs[i + 1].try_into().unwrap()
+                        };
+                        sigs_vec.push((preimage, digit_arr[0]));
+                    }
+                    sigs_vec.try_into().unwrap()
+                }
+
+                /// Generate a WOTS public key using the provided secret.
+                pub fn generate_public_key(secret: &str) -> PublicKey {
+                    let secret_key = match hex::decode(secret) {
+                        Ok(bytes) => bytes,
+                        Err(_) => panic!("Invalid hex string for secret"),
+                    };
+                    let params = winternitz::Parameters::new(MSG_LEN * 2, 4);
+                    let pubkey_vec = winternitz::generate_public_key(&params, &secret_key);
+                    pubkey_vec.try_into().unwrap()
+                }
+
+                /// A sub-module for the compact signature variant.
+                pub mod compact {
+                    use super::*;
+
+                    /// The compact signature is just the 20-byte preimages.
+                    pub type Signature = [[u8; 20]; N_DIGITS as usize];
+
+                    /// Create a verification script for the compact WOTS public key.
+                    pub fn checksig_verify(public_key: PublicKey) -> Script {
+                        let params = winternitz::Parameters::new(MSG_LEN * 2, 4);
+                        WINTERNITZ_MESSAGE_COMPACT_VERIFIER.checksig_verify(&params, &public_key.to_vec())
+                    }
+
+                    /// Generate a compact signature for a message.
+                    pub fn get_signature(secret: &str, msg_bytes: &[u8]) -> Signature {
+                        let secret_key = match hex::decode(secret) {
+                            Ok(bytes) => bytes,
+                            Err(_) => panic!("Invalid hex string for secret"),
+                        };
+
+                        let sigs = WINTERNITZ_MESSAGE_COMPACT_VERIFIER.sign(
+                            &winternitz::Parameters::new(MSG_LEN * 2, 4),
+                            &secret_key,
+                            &msg_bytes.to_vec(),
+                        );
+                        assert_eq!(sigs.len(), N_DIGITS as usize);
+
+                        let mut sigs_vec: Vec<[u8; 20]> = Vec::new();
+                        // Iterate over the signature pieces using step_by.
+                        for i in (0..sigs.len()).step_by(2) {
```

Comment:
should be step_by(1) because compact signature doesn't have interleaving messages

Reply:
wots_compact::get_signature() isn't used by any other module. so this bug doesn't affect existing function.

Reply:
added fix for this